front/app/containers/Admin/projects/all/index.tsx

```diff
@@ -57,11 +53,14 @@ export interface Props {
 
 const AdminProjectsList = memo(({ className }: Props) => {
   const { data: authUser } = useAuthUser();
-  const userIsAdmin = !isNilOrError(authUser) ? isAdmin(authUser) : false;
-  const userIsFolderModerator = !isNilOrError(authUser)
-    ? isProjectFolderModerator(authUser.data)
-    : false;
   const isProjectFoldersEnabled = useFeatureFlag({ name: 'project_folders' });
+  const userIsAdmin = isAdmin(authUser) ?? false;
+  const userIsFolderModerator =
+    (authUser &&
+      isProjectFoldersEnabled &&
+      isProjectFolderModerator(authUser.data)) ??
+    false;
+  const userCanCreateProject = userIsAdmin || userIsFolderModerator;
```

Comment:
Yes, this is a really good point. I'll look into streamlining this, instead of having checks in the component. We are also taking a feature flag into account here but maybe it's not so bad to also consider this in the function.